File: programme/area-of-right-angle-triangle/area-of-right-angle-triangle.c
/**
 * Computes the area of a right angle triangle
 */

#include <stdio.h>
#include <assert.h>         // for the assert function

double area_of_right_angle_triangle(double base, double height);
void test(void);

int main(void) {
    double area, base, height;

    test();

    printf("base: ");
    scanf("%lf", &base);
    printf("height: ");
    scanf("%lf", &height);
    area = area_of_right_angle_triangle(base, height);
    printf("area: %.1lf\n", area);

    return 0;
}

// calculates the area
double area_of_right_angle_triangle(double base, double height) {
    return 0.5 * base * height;
}

// checks correctness of the area calculation
void test(void) {
    assert(area_of_right_angle_triangle(2, 3) == 3);
}

File: programme/area-of-right-angle-triangle/area-of-right-angle-triangle.php
<?php
// Write a programme to find the area of a right angle triangle
// Formula: area = 1/2 * base * height

function calculate_area(int | float $base, int | float $height)
{

    if (!is_numeric($base)) {

        return 'Invalid input: Base is not a number';

    }

    if (!is_numeric($height)) {

        return 'Invalid input: Height is not a number';

    }

    $area = ($base * $height) / 2;
    return $area;
}

$base = readline('Enter the base   : ');
$height = readline('Enter the height : ');
$area = calculate_area($base, $height);
print "\nBase   : $base";
print "\nHeight : $height";
print "\nCalculated Area : $area\n";
